chia/server/server.py

```diff
@@ -689,6 +689,9 @@ def accept_inbound_connections(self, node_type: NodeType) -> bool:
     def is_trusted_peer(self, peer: WSChiaConnection, trusted_peers: Dict) -> bool:
         if trusted_peers is None:
             return False
+        if peer.peer_host == "127.0.0.1":
```

Comment:
I don't think this is safe. you can have other (untrusted) user accounts on your machine, but perhaps most importantly, you may be running untrusted javascript code in a browser on localhost.

I'm not sure about the surrounding code here, but this check looks like it's happening before validating the certificate of the peer, in which case it would circumvent our protection against malicious web sites.

Reply:
perhaps the threat of trusting a malicious node here isn't that great. This node still can't sign transactions, it can just lie about the state of the blockchain, right?

Reply:
also, technically, to check whether the other end is localhost, we would check against `127.0.0.0/8` and `::1/128`

Reply:
i add it because thats how we are doing things right now for fetch_blocks_and_validate, my motivation is for all methods to use the same "trusted_peer_logic" and i dident want to break the current flow but i guess its unnecessary as the certs should match anyway